lib/Mojo/Collection.pm

```diff
@@ -304,6 +323,32 @@ C<$b> will always be set to the next element in the collection.
 
 Create a new collection with all elements in reverse order.
 
+=head2 pop
+
+  # [1, 2, 3, 4]
+  my $new = c(1 .. 5)->pop;
```

Comment:
I think this is very confusing that the variable is called `$new`. Sounds like it's not mutating, even though the documentation says so.